### PR TITLE
Add an example docker-compose file

### PR DESCRIPTION
**Pull request by thechadraccoon**

### CODE REVIEW

File: example.docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    Environment: 
+      - 'PUFFER_LOGS=/etc/pufferpanel/logs', # Location of your logs
+      - 'PUFFER_PANEL_DATABASE_DIALECT=sqlite3', # Databse type https://docs.pufferpanel.com/en/2.x/guides/database.html
+      - 'PUFFER_PANEL_DATABASE_URL=file:/etc/pufferpanel/pufferpanel.db?cache=shared', # SQL URL https://docs.pufferpanel.com/en/2.x/guides/da>
+      - 'PUFFER_DAEMON_DATA_CACHE=/var/lib/pufferpanel/cache', # Cache location within the container
+      - 'PUFFER_DAEMON_DATA_SERVERS=/var/lib/pufferpanel/servers', # Server data location within the container
+      - 'PUFFER_DAEMON_DATA_MODULES=/var/lib/pufferpanel/modules', # Location for Modules within the container
+
+    Volumes:
+      - '/etc/pufferpanel:/etc/pufferpanel', # Config/logs folder
```
Comment (LordRalex) [inline example.docker-compose.yml:16]: Use a volume for the config directory instead of the host.

File: example.docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Version: '3'
+Service:
+  Pufferpanel:
```
Comment (LordRalex) [inline example.docker-compose.yml:3]: Should be using lowercase naming for all keys

File: example.docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Image: pufferpanel/pufferpanel
+    Container_name: Pufferpanel
```
Comment (LordRalex) [inline example.docker-compose.yml:5]: Name is not used by compose anymore

**Review by LordRalex — CHANGES_REQUESTED** [review]

File: example.docker-compose.yml (hunk 3 above)
Comment (thechadraccoon) [reply example.docker-compose.yml:5]: https://docs.docker.com/compose/compose-file/compose-file-v3/
Simply untrue.

File: example.docker-compose.yml (hunk 2 above)
Comment (thechadraccoon) [reply example.docker-compose.yml:3]: Preference, makes no functional difference.

File: example.docker-compose.yml (hunk 1 above)
Comment (thechadraccoon) [reply example.docker-compose.yml:16]: Documentation sake. Avoids maintaining separate version of docs for depending on installation method.

File: example.docker-compose.yml (hunk 2 above)
Comment (LordRalex) [reply example.docker-compose.yml:3]: This is not a preference. Yaml is case sensitive. Docker might not care, but it is not correct, not does it match their docs. Use lowercase keys.

File: example.docker-compose.yml (hunk 3 above)
Comment (LordRalex) [reply example.docker-compose.yml:5]: It is true, if swarm mode is used. May not always be true, but it is also not wrong in cases.

File: example.docker-compose.yml (hunk 1 above)
Comment (LordRalex) [reply example.docker-compose.yml:16]: We already use docker volumes for the config when plain docker is used. This should essentially match those steps.

**Comment by oskardotglobal** [comment]
There aren't supposed to be any commas after list elements...

File: example.docker-compose.yml (hunk 1 above)
Comment (thechadraccoon) [reply example.docker-compose.yml]: switched to volumes.

File: example.docker-compose.yml (hunk 2 above)
Comment (thechadraccoon) [reply example.docker-compose.yml]: keys made lowercase.

**Comment by thechadraccoon** [comment]
> There aren't supposed to be any commas after list elements...
That's my bad, I made this file from my personnal docker-compose file. I use ['value1','value2'] format for lists.

**Comment by oskardotglobal** [comment]
Just checked it again, it's looking good. The only thing I see is the config volume hasn't the most clear name and since compose volumes are named `<project name>_<volume>` the volumes will probably end up being named `pufferpanel_pufferpanel` and `pufferpanel_pufferpanel-data`, so it'd probably be more reasonable to name them something like `config` and `data`

File: example.docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      - 'pufferpanel:/etc/pufferpanel' # Config/logs folder
+      - 'pufferpanel-data:/var/lib/pufferpanel' # Data folder
+      - '/var/run/docker.sock:/var/run/docker.sock' # Docker socket, leave this in.
+    ports:
+      - '8080:8080' # Panel port
+      - '5657:5657' # SFTP port
+      - '27015:27015' # Default SRCDS port
+      - '25565:25565' # Default Minecraft Java port
+volumes:
+  - pufferpanel:
```
[the remark this answers is not in the document]
Comment (oskardotglobal) [reply example.docker-compose.yml:24]: Yes, I didn't see that. I'd be better to leave it out

File: example.docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+version: '3'
+service:
```
Comment (MaxJa4) [inline example.docker-compose.yml:2]: ```services```
Otherwise invalid file and 'Additional property service is not allowed' error appears.

**Comment by rlnt** [comment]
I'd also like to see UID and GID environment variables to set the user and the group for the process.

**Comment by Tenta-dev** [comment]
I tried this one and that worked for me : 
```yaml
version: '3'
services:
  pufferpanel:
    image: pufferpanel/pufferpanel
    container_name: Pufferpanel
    restart: on-failure
    environment: 
      - 'PUFFER_LOGS=/etc/pufferpanel/logs' # Location of your logs
      - 'PUFFER_PANEL_DATABASE_DIALECT=sqlite3' # Databse type https://docs.pufferpanel.com/en/2.x/guides/database.html
      - 'PUFFER_PANEL_DATABASE_URL=file:/etc/pufferpanel/pufferpanel.db?cache=shared' # SQL URL https://docs.pufferpanel.com/en/2.x/guides/database.html
      - 'PUFFER_DAEMON_DATA_CACHE=/var/lib/pufferpanel/cache' # Cache location within the container
      - 'PUFFER_DAEMON_DATA_SERVERS=/var/lib/pufferpanel/servers' # Server data location within the container
      - 'PUFFER_DAEMON_DATA_MODULES=/var/lib/pufferpanel/modules' # Location for Modules within the container
    volumes:
      - './config:/etc/pufferpanel' # Config/logs folder
      - './pufferpanel-data:/var/lib/pufferpanel' # Data folder
      - '/var/run/docker.sock:/var/run/docker.sock' # Docker socket, leave this in.
    ports:
      - '8080:8080' # Panel port
      - '5657:5657' # SFTP port
      - '25565:25565' # Default Minecraft Java port
```

**Pull request closed without merging**